zerver/tests/test_signup.py

```diff
@@ -3261,17 +3261,30 @@ def signup(self, *,
         email = "newguy@zulip.com": str,
         password = "newpassword": str,
         realm=get_realm("zulip"): Realm,
-        subdomain=None: Optional[str]) -> UserProfile:
+        subdomain=None: Optional[str],
+        full_name=None: Optional[str],
+        # These params stop the signup early and return the last result,
+        # to allow the caller to check for specific, expected exceptions
+        stop_after_reg_form=False: bool) -> Union[UserProfile, HttpResponse]:
 
         result = self.client_post("/accounts/home/", {"email": email}, subdomain=subdomain)
         self.assertEqual(result.status_code, 302)
+        self.assertTrue(result["Location"].endswith(f"/accounts/send_confirm/{email}"))
         result = self.client_get(result["Location"], subdomain=subdomain)
+        self.assert_in_response("Check your email so we can get started.", result)
 
+        # Visit the confirmation link.
```

Comment:
I'd try to put the basic comments and extra assertions in the first commit of a `verify_signup` function, so that later commits just add features needed for the test being ported, and not generically useful verification.

Reply:
OK